### PR TITLE
TreeSyntax: use input Lit.{Char,String} if present

### LINKED ISSUE #1384: Lit.String and Lit.Char format is not preserved
Obtained:

```scala
import scala.meta._
"\"\\n\"".parse[Term].get.asInstanceOf[Lit.String].value
res31: String =
"
"
"\'\\n\'".parse[Term].get.asInstanceOf[Lit.Char].value
res33: Char =


```

Expected:

```scala
import scala.meta._
"\"\\n\"".parse[Term].get.asInstanceOf[Lit.String].value == "\\n"
"\'\\n\'".parse[Term].get.asInstanceOf[Lit.Char].value == "\\n"
```

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #1384.

### CODE REVIEW

File: scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala [hunk 1]
```diff
@@ -819,13 +819,19 @@ object TreeSyntax {
...
-        m(
-          Literal,
-          s(enquote(value.toString, if (value.contains("\n")) TripleQuotes else DoubleQuotes))
-        )
+      case t @ Lit.Char(value) =>
+        val syntax = t.pos match {
+          case Position.None => enquote(value.toString, SingleQuotes)
+          case pos => pos.text
+        }
+        m(Literal, s(syntax))
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:827]: do you mean everything, or just the Lit.Char part?

[the remark this answers is not in the document]
Comment (dos65) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:827]: yep - only `List.Char` part changes

[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala:827]: done, @dos65 , ptal. discovered #2730 when trying to make the tests work.

**Review by dos65 — APPROVED** [review]
LGTM!

**Pull request merged**